bindings/rust/extended/s2n-tls/src/callbacks.rs

```diff
@@ -48,12 +48,12 @@ pub use pkey::*;
 /// callbacks were configured through the Rust bindings.
 pub(crate) unsafe fn with_context<F, T>(conn_ptr: *mut s2n_connection, action: F) -> T
 where
-    F: FnOnce(&mut Connection, &mut Context) -> T,
+    F: FnOnce(&mut Connection, &Context) -> T,
 {
     let raw = NonNull::new(conn_ptr).expect("connection should not be null");
     let mut conn = Connection::from_raw(raw);
-    let mut config = conn.config().expect("config should not be null");
-    let context = config.context_mut();
```

Comment:
Can we remove the `context_mut` method entirely? It should really only be used during the builder. If it's on the config, it's too easy to call. Or at the very least mark it `unsafe`.